pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -187,6 +188,32 @@ func (p *Predictor) Reconcile(ctx context.Context, isvc *v1beta1.InferenceServic
 		if kstatus, err = p.reconcileKnativeDeployment(ctx, isvc, &objectMeta, &podSpec); err != nil {
 			return ctrl.Result{}, err
 		}
+
+		forceStopRuntime := "false"
+		if val, exist := isvc.Annotations[constants.StopAnnotationKey]; exist {
+			forceStopRuntime = val
+		}
+
+		if strings.EqualFold(forceStopRuntime, "true") {
+			// Clear all statuses if the ISVC is stopped
```

Comment:
Is there a reason for doing this in the predictor rather than the top level controller? IIRC all the existing logic for setting isvc statuses is in the [top level controller](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/controller.go#L297). I think it would be better to be consistent and move this logic there unless I'm missing something.